firedrake/slate/slac/compiler.py

```diff
@@ -136,14 +136,16 @@ def compile_expression(slate_expr, tsfc_parameters=None):
 
             # If tensor is mixed, there will be more than one SplitKernel
             incl = []
+            split_forms = cxt_kernel.form_indices_map
             for splitkernel in cxt_kernel.tsfc_kernels:
                 index = splitkernel.indices
                 kinfo = splitkernel.kinfo
 
                 # Generate an iterable of coefficients to pass to the subkernel
-                # if any are required
-                clist = [c for ci in kinfo.coefficient_map
-                         for c in builder.coefficient(exp.coefficients()[ci])]
+                # if any are required.
+                sform = split_forms[index]
+                clist = [c for i in kinfo.coefficient_map
+                         for c in builder.coefficient(sform.coefficients()[i])]
 
```

Comment:
The `ContextKernel` contains the relevant mapping between the `kinfo.indices` and the associated split form. Then the `kinfo.coefficient_map` provides the map to the relevant coefficients of the split form. This way, the local assembly kernel should have the right coefficient arguments.